flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,278 @@
+    private long batchOffsetInFile;
+
+    LongOrcColumnVector(long batchOffsetInFile) {
+      this.batchOffsetInFile = batchOffsetInFile;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return false;
+    }
+
+    @Override
+    public long getLong(int i) {
+      return batchOffsetInFile + i;
+    }
+  }
+
+  private static class StructConverter implements Converter {
+    private final Types.StructType structType;
+    private final List<Converter> fieldConverters;
+    private final Map<Integer, ?> idToConstant;
+
+    private StructConverter(Types.StructType structType, List<Converter> fieldConverters,
+                            Map<Integer, ?> idToConstant) {
+      this.structType = structType;
+      this.fieldConverters = fieldConverters;
+      this.idToConstant = idToConstant;
+    }
+
+    @Override
+    public ColumnVector convert(org.apache.orc.storage.ql.exec.vector.ColumnVector vector, int batchSize,
+                                long batchOffsetInFile) {
+      StructColumnVector structVector = (StructColumnVector) vector;
+      List<Types.NestedField> fields = structType.fields();
+      ColumnVector[] fieldVectors = new ColumnVector[fields.size()];
+      for (int pos = 0; pos < fields.size(); pos++) {
+        Types.NestedField field = fields.get(pos);
+        if (idToConstant.containsKey(field.fieldId())) {
+          fieldVectors[pos] = getConstantVectors(field.type(), idToConstant.get(field.fieldId()));
+        } else if (field.equals(MetadataColumns.ROW_POSITION)) {
+          fieldVectors[pos] = new LongOrcColumnVector(batchOffsetInFile);
+        } else {
+          fieldVectors[pos] = fieldConverters.get(pos)
+              .convert(structVector.fields[pos], batchSize, batchOffsetInFile);
+        }
+      }
+
+      return new FlinkRowColumnVector(fieldVectors, structVector);
+    }
+
+    private ColumnVector getConstantVectors(Type type, Object constant) {
+      if (type.isPrimitiveType()) {
+        Type.TypeID typeID = type.typeId();
+        switch (typeID) {
+          case INTEGER:
+          case DATE:
+          case TIME:
+            return ConstantColumnVectors.ints(constant);
+
+          case LONG:
+            return ConstantColumnVectors.longs(constant);
+
+          case BOOLEAN:
+            return ConstantColumnVectors.booleans(constant);
+
+          case DOUBLE:
+            return ConstantColumnVectors.doubles(constant);
+
+          case FLOAT:
+            return ConstantColumnVectors.floats(constant);
+
+          case DECIMAL:
+            return ConstantColumnVectors.decimals(constant);
+
+          case TIMESTAMP:
+            return ConstantColumnVectors.timestamps(constant);
+
+          case STRING:
+            return ConstantColumnVectors.bytes(constant);
+
+          default:
+            throw new UnsupportedOperationException("Unsupported data type for constant.");
```

Comment:
For the iceberg data type `UUID`, `FIXED`, `BINARY` cases,  we should return `ConstantColumnVectors.bytes(constant)`.    The correct mapping relationships (iceberg data types <=> flink data types <=> orc data types)  are [here](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcReader.java#L87).